test/model/src/core/kv_database.cpp

```diff
@@ -102,8 +102,13 @@ kv_database::create_checkpoint_nolock(
             throw model_exception("Checkpoint already exists: " + ckpt_name);
     }
 
+    /* Remember the oldest timestamp only if the stable timestamp is set. */
```

Comment:
That's an interesting case, is that what WT does as well?

Reply:
Yes. I don't know if it's documented, but empirically I see that this is what happens. You can check that the unit tests verify that WT has the same behavior. But I agree that it is somewhat unexpected!

Reply:
It has been confirmed that this is how WT works. Whether it is the right behaviour or not, it is being discussed within the team. The outcome is beyond the scope of this PR - resolving.